scripts/send-email-to-pre-fxa-subscribers.js

```diff
@@ -8,42 +10,52 @@ const EmailHelpers = require("../template-helpers/emails.js");
 const EmailUtils = require("../email-utils");
 const { LocaleUtils } = require ("../locale-utils");
 
+const PAGE_SIZE = process.env.PAGE_SIZE;
+
 
 (async (req) => {
   const localeUtils = LocaleUtils.init();
   EmailUtils.init();
-
-  const subscribers = await DB.getPreFxaSubscribers();
-  console.log(`Found ${subscribers.length} subscriber records with empty fxa_uid.`);
   const notifiedSubscribers = [];
   const utmID = "pre-fxa";
-  for (const subscriber of subscribers) {
-    const signupLanguage = subscriber.signup_language;
-    const subscriberEmail = subscriber.primary_email;
-    const requestedLanguage = signupLanguage ? acceptedLanguages(signupLanguage) : "";
-    const supportedLocales = negotiateLanguages(
-      requestedLanguage,
-      localeUtils.availableLanguages,
-      {defaultLocale: "en"}
-    );
-
-    if (!notifiedSubscribers.includes(subscriberEmail)) {
-      const sendInfo = await EmailUtils.sendEmail(
-        subscriberEmail,
-        LocaleUtils.fluentFormat(supportedLocales, "pre-fxa-subject"), // email subject
-        "default_email", // email template
-        {
-          supportedLocales,
-          SERVER_URL: AppConstants.SERVER_URL,
-          unsubscribeUrl: EmailUtils.getUnsubscribeUrl(subscriber, utmID), // need to test the flow for legacy users who want to unsubscribe
-          ctaHref: EmailHelpers.getPreFxaUtmParams(AppConstants.SERVER_URL, "create-account-button", subscriberEmail),
-          whichPartial: "email_partials/pre-fxa",
-          preFxaEmail: true,
-          email: subscriberEmail,
-        },
+
+  const subscribersResult = await DB.getPreFxaSubscribersPage({ perPage: PAGE_SIZE, currentPage: 1 });
+  console.log(`Found ${subscribersResult.pagination.total} subscriber records with empty fxa_uid.`);
+  console.log(`Will process ${subscribersResult.pagination.lastPage} pages of size ${PAGE_SIZE}.`);
+  const lastPage = subscribersResult.pagination.lastPage;
+
+
+  for (let currentPage = 1; currentPage <= lastPage; currentPage++) {
+    console.log(`Processing page ${currentPage} of ${lastPage}.`);
+    const subscribersPageResult = await DB.getPreFxaSubscribersPage({ perPage: PAGE_SIZE, currentPage });
```

Comment:
This line looks the same as the one above on L22 (both fetching `currentPage=1`). But the first usage seems to just get stats, so I don't think we're processing anything twice or whatever. I think.

Reply:
Yeah first usage is to get the `lastPage` number to put into the `for` loop. But trying to write the `for` loop to accommodate the first-page-is-already-queried case made the logic more complicated.